### ISSUE TITLE
[multiple lessons] Prioritization of content / episodes

### ISSUE THREAD

**Issue by bsmith89** [swcarpentry/curriculum-advisors#3]
Most (all?) of the SWC lessons take longer than 3 hours (or 6 for e.g. Python/R) to teach to a workshop with beginners. This means that, for a traditional class schedule, each lesson needs to be heavily culled.

The problem:

- This is not well explained to first-time instructors
- can be demotivating when they don't make it all the way through, and
- no suggested prioritization of the material is clearly communicated.

Potential fixes:

- The CAC could request that maintainers prepare and clearly link prioritization instructions for instructors. 
- We could also suggest a prioritization itself.
- Finally, we can ask for (or suggest) more accurate run times for the episodes within lessons.

Relevant references (that I'm currently aware of):

- swcarpentry/shell-novice#928
- swcarpentry/python-novice-inflammation#452
- In the [lesson contribution instructions](https://github.com/carpentries/workshop-template/blob/gh-pages/CONTRIBUTING.md#what-not-to-contribute)
- The [shell-novice instructor notes](https://swcarpentry.github.io/shell-novice/guide/index.html#preparing-to-teach)

Potential downsides / complications:

- Not every instructor or workshop takes the same amount of time on the episodes. Time estimates may be less accurate than would be helpful
- Some instructors may have differing opinions about material prioritization, or some audiences may have differing levels of experience; where would this prioritization fall on the scale between "a helpful suggestion" and "mandatory"?

**Comment by richmccue** [comment]
I mentioned this on the survey thread, but another approach might be to disaggregate workshop activities to make it easier for instructors to customize their particular session to better meet the needs of participants and/or respond to time constraints.

**Comment by bsmith89** [comment]
Yes, I can definitely see the connection. In many of the topics we'll discuss as a committee I think there's a general theme regarding the spectrum between between two extremes:

- a highly curated, standardized, and integrated curriculum, versus
- a customizable, bite-size, and flexible set of episodes or lessons

Not sure these are mutually exclusive, but many of our suggestions will have to navigate some sort of trade-off between them.

**Comment by richmccue** [comment]
Toby, I agree that identifying learning objectives is extremely helpful in creating workshops that focus on the important or key concepts and skills. Here's an example of some learning objectives from the UVic Libraries' Data Analysis with Excel workshop: https://uviclibraries.github.io/excel/

Just in case it's helpfule, here is our "Workshop, Workshop" that all the interns and graduate students we hire go through to introduce them to best practices for creating and modifying our library's digital fluency workshops. It includes a module on creating learning objectives, and has it's own learning objectives on this page: https://uviclibraries.github.io/hyflex-active-workshops/ 

I hope his helps.

**Comment by bsmith89** [comment]
Are the learning objectives we're discussing here different from the "Objectives" at the top of each episode?

I was under the impression that the objective setting work was already done for many of these lessons. ...perhaps it has gotten a tad bloated now, though.

**Comment by bsmith89** [comment]
Thank you very much for your valuable input, Erin! We'll certainly consider this as a potential course of action.

Would committee members (or other folks, of course) like to propose any similarly concrete plans? I imagine explicit options like these will make our discussion much more effective going forward.

**Comment by richmccue** [comment]
In the short run, Erin's suggestion makes a lot of sense and I fully support it. In the long run, encouraging workshop maintainers to think about and create specific learning objectives for their workshops (with support and training from CAC) would align with best practices in the teaching and learning literature for workshop or lesson plan creation. This should be helpful for both instructors and learners to know exactly what skills learners should be able to demonstrate upon successful workshop completion. The workshop schedule descriptions would be a good starting point and probably be the basis for the generation of workshop learning objectives.

Given this discussion, I'd argue that it would probably be helpful if maintainers eventually identified core and stretch learning objectives, with successful workshop completion dependent on demonstrating competency of the core learning objectives, and stretch learning objectives for those who would like to learn additional optional skills. This would allow for the addition of new optional activities to workshops that could be used by instructors if desired depending on the interests and goals of the organization hosting the workshop or of individual learners.

**Comment by richmccue** [comment]
I can't remember if it was a suggestion for this Issue or another one, but as promised here are some resources for how to use the Hypothes.is as a tool for collaborative web annotation that might be helpful in facilitating threaded discussions on workshop web pages for possible updates to workshops:
- Group annotation video: https://www.youtube.com/watch?v=VfojpdGhSSI
- Step-by-step Hypothesis groups how-to: https://web.hypothes.is/help/annotating-with-groups/
